internal/cri/store/container/status.go

```diff
@@ -164,21 +166,54 @@ type StatusStorage interface {
 // StoreStatus creates the storage containing the passed in container status with the
 // specified id.
 // The status MUST be created in one transaction.
+// If atomic writing fail due to no space, try use the in-place modification method, see issue 7247 for details.
 func StoreStatus(root, id string, status Status) (StatusStorage, error) {
 	data, err := status.encode()
 	if err != nil {
 		return nil, fmt.Errorf("failed to encode status: %w", err)
 	}
 	path := filepath.Join(root, "status")
 	if err := continuity.AtomicWriteFile(path, data, 0600); err != nil {
-		return nil, fmt.Errorf("failed to checkpoint status to %q: %w", path, err)
+		if strings.Contains(err.Error(), "no space") {
+			err = writeFileInplace(path, data, 0600)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to checkpoint status to %q: %w", path, err)
+		}
 	}
 	return &statusStorage{
 		path:   path,
 		status: status,
 	}, nil
 }
 
+// writeFileInplace writes the data to the file in-place.
+func writeFileInplace(path string, data []byte, perm os.FileMode) error {
```

Comment:
What is "in-place"? How does this differ from `os.WriteFile`?

Reply:
Why can't we just use `os.WriteFile` here?

Reply:
could use `file descripter` execute `Sync` to make sure had saved on disk, try to minimize the time that may cause inconsistent JSON struct content in the file

Reply:
not update status file when root directory full